File: 1637-widest-vertical-area-between-two-points-containing-no-points/1637-widest-vertical-area-between-two-points-containing-no-points.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        points = [i[0] for i in points]
        points.sort()
        m = 0
        for i in range(len(points[1:])):
            m = max(m, points[i] - points[i - 1])
        return m

File: 1637-widest-vertical-area-between-two-points-containing-no-points/1637-widest-vertical-area-between-two-points-containing-no-points.rs
impl Solution {
    pub fn max_width_of_vertical_area(points: Vec<Vec<i32>>) -> i32 {
        let mut v1:Vec<i32> = vec![0;points.len()];
        for i in 0..points.len(){
            v1[i]=points[i][0];
        }
        v1.sort();
        let mut max = 0;
        for i in 1..v1.len(){
            if v1[i] - v1[i-1] > max{
                max = v1[i] - v1[i-1];
            }
        }
        return max
    }
}